pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -190,31 +204,43 @@ func buildHTTPVirtualServices(obj config.Config, gateways map[parentKey]map[k8s.
 
 	httproutes := []*istio.HTTPRoute{}
 	hosts := hostnameToStringList(route.Hostnames)
-	for _, r := range route.Rules {
+	for i := 0; i < len(route.Rules); i++ {
+		r := route.Rules[i]
+		if len(r.Matches) > 1 {
+			// split the rule to make sure each rule has up to one match
+			for _, match := range r.Matches {
+				splitRule := *r.DeepCopy()
+				splitRule.Matches = []k8s.HTTPRouteMatch{match}
+				route.Rules = append(route.Rules, splitRule)
+			}
+			continue
+		}
 		// TODO: implement rewrite, timeout, mirror, corspolicy, retries
 		vs := &istio.HTTPRoute{}
 		for _, match := range r.Matches {
 			uri, err := createURIMatch(match)
 			if err != nil {
 				reportError(err)
-				return nil
+				return
 			}
 			headers, err := createHeadersMatch(match)
 			if err != nil {
 				reportError(err)
-				return nil
+				return
 			}
 			qp, err := createQueryParamsMatch(match)
 			if err != nil {
 				reportError(err)
-				return nil
+				return
 			}
 			method, err := createMethodMatch(match)
 			if err != nil {
 				reportError(err)
-				return nil
+				return
 			}
 			vs.Match = append(vs.Match, &istio.HTTPMatchRequest{
+				// the name is a hack to fetch the original path type
+				Name:        generateMatchName(obj, i, *match.Path.Type),
```

Comment:
why do we need Type? isn't index sufficient?

Reply:
Because we are using `regex` to implement both `prefix` and `regex`, if we don't have the type, we would not be able to determine whether to trim the regex suffix for `prefix` when calculating the uri length

For example:
https://github.com/istio/istio/pull/37139/files/8cb067ba9c70430f11a7ef942b4cf9e19a640915#diff-13176e82f5f2350940ea3ea83d0bfeb3147c172af5808771f6006be8393e1632R33-R45

The index is to make match names not duplicated

Reply:
ah I didn't realize we were later parsing it in getOriginalPathType. Hmm.. it seems more ideal if we can have it somewhere else instead of parsing but that may not be possible